examples/reflink_block.rs

```diff
@@ -0,0 +1,34 @@
+use std::fs::File;
+use std::num::NonZeroU64;
+
+fn main() -> std::io::Result<()> {
+    let args: Vec<_> = std::env::args().collect();
+    if args.len() < 4 {
+        eprintln!(
+            "Usage: {} <source_file> <target_file> <cluster_size>",
+            args[0]
+        );
+        return Ok(());
+    }
+    let src_file = &args[1];
+    let tgt_file = &args[2];
+    let cluster_size =
+        NonZeroU64::new(args[3].parse().expect("cannot parse cluster size")).unwrap();
```

Comment:
IMO let-else here is shorter and `NonZeroU64` can be parsed directly, but just a nit and feel free to ignore it:

```suggestion
    let args: Vec<_> = std::env::args().collect();
    
    let [_, src_file, tgt_file, cluster_size] = &args else {
        eprintln!(
            "Usage: {} <source_file> <target_file> <cluster_size>",
            args[0]
        );
        return Ok(())
    };

    let cluster_size: NonZeroU64 =
        cluster_size.parse().expect("cannot parse cluster size");
```

Reply:
That looks better for sure, thanks for the tip :)

Reply:
Updated in the latest commit